Increase timeout for API calls

### Expected Behavior

Reproducer:

```yaml
id: openai
namespace: dev

tasks:
  - id: prompt
    type: io.kestra.plugin.openai.ChatCompletion
    apiKey: "{{ secret('OPENAI_API_KEY') }}"
    model: gpt-4
    prompt: Why do data engineers build data pipelines? Give examples of common types of data pipelines
```

More complex prompts (here we ask for 2 things: explain something + give examples) require longer text responses and take longer to process. 

### Actual Behaviour

```
java.net.SocketTimeoutException: timeout
timeout
java.lang.RuntimeException: java.net.SocketTimeoutException: timeout
	at io.reactivex.internal.util.ExceptionHelper.wrapOrThrow(ExceptionHelper.java:46)
	at io.reactivex.internal.observers.BlockingMultiObserver.blockingGet(BlockingMultiObserver.java:93)
	at io.reactivex.Single.blockingGet(Single.java:2870)
	at com.theokanning.openai.service.OpenAiService.execute(OpenAiService.java:317)
	at com.theokanning.openai.service.OpenAiService.createChatCompletion(OpenAiService.java:131)
	at io.kestra.plugin.openai.ChatCompletion.run(ChatCompletion.java:156)
	at io.kestra.plugin.openai.ChatCompletion.run(ChatCompletion.java:25)
	at io.kestra.core.runners.Worker$WorkerThread.run(Worker.java:641)
Caused by: java.net.SocketTimeoutException: timeout
	at okhttp3.internal.http2.Http2Stream$StreamTimeout.newTimeoutException(Http2Stream.java:678)
	at okhttp3.internal.http2.Http2Stream$StreamTimeout.exitAndThrowIfTimedOut(Http2Stream.java:686)
	at okhttp3.internal.http2.Http2Stream.takeHeaders(Http2Stream.java:154)
	at okhttp3.internal.http2.Http2ExchangeCodec.readResponseHeaders(Http2ExchangeCodec.java:136)
	at okhttp3.internal.connection.Exchange.readResponseHeaders(Exchange.java:115)
	at okhttp3.internal.http.CallServerInterceptor.intercept(CallServerInterceptor.java:94)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:142)
	at okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.java:43)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:142)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:117)
	at okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.java:94)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:142)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:117)
	at okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.java:93)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:142)
	at okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.java:88)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:142)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:117)
	at com.theokanning.openai.service.AuthenticationInterceptor.intercept(AuthenticationInterceptor.java:28)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:142)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:117)
	at okhttp3.RealCall.getResponseWithInterceptorChain(RealCall.java:229)
	at okhttp3.RealCall.execute(RealCall.java:81)
	at retrofit2.OkHttpCall.execute(OkHttpCall.java:204)
	at retrofit2.adapter.rxjava2.CallExecuteObservable.subscribeActual(CallExecuteObservable.java:46)
	at io.reactivex.Observable.subscribe(Observable.java:12284)
	at retrofit2.adapter.rxjava2.BodyObservable.subscribeActual(BodyObservable.java:35)
	at io.reactivex.Observable.subscribe(Observable.java:12284)
	at io.reactivex.internal.operators.observable.ObservableSingleSingle.subscribeActual(ObservableSingleSingle.java:35)
	at io.reactivex.Single.subscribe(Single.java:3666)
	at io.reactivex.Single.blockingGet(Single.java:2869)
	... 5 more
```

### Steps To Reproduce

_No response_

### Environment Information

- Kestra Version: 0.11.0 develop full


### Example flow

_No response_

--- a/src/main/java/io/kestra/plugin/openai/AbstractTask.java
+++ b/src/main/java/io/kestra/plugin/openai/AbstractTask.java
@@ -4,11 +4,10 @@ import com.theokanning.openai.service.OpenAiService;
 import io.kestra.core.exceptions.IllegalVariableEvaluationException;
 import io.kestra.core.models.tasks.Task;
 import io.kestra.core.runners.RunContext;
-import lombok.EqualsAndHashCode;
-import lombok.Getter;
-import lombok.NoArgsConstructor;
-import lombok.ToString;
+import lombok.*;
 import lombok.experimental.SuperBuilder;
+
+import java.time.Duration;
 
 @SuperBuilder
 @ToString
@@ -20,9 +19,12 @@ public abstract class AbstractTask extends Task implements OpenAiInterface{
 
     protected String user;
 
+    @Builder.Default
+    protected long clientTimeout = 10;
+
     protected OpenAiService client(RunContext runContext) throws IllegalVariableEvaluationException {
         String apiKey = runContext.render(this.apiKey);
 
-        return new OpenAiService(apiKey);
+        return new OpenAiService(apiKey, Duration.ofSeconds(clientTimeout));
     }
 }

--- a/src/main/java/io/kestra/plugin/openai/OpenAiInterface.java
+++ b/src/main/java/io/kestra/plugin/openai/OpenAiInterface.java
@@ -18,4 +18,9 @@ public interface OpenAiInterface {
     )
     @PluginProperty(dynamic = true)
     String getUser();
+
+    @Schema(
+        title = "The maximum number of seconds to wait for a response."
+    )
+    long getClientTimeout();
 }

--- a/src/test/java/io/kestra/plugin/openai/ChatCompletionTest.java
+++ b/src/test/java/io/kestra/plugin/openai/ChatCompletionTest.java
@@ -11,7 +11,8 @@ import org.junit.jupiter.api.Test;
 import java.util.List;
 
 import static org.hamcrest.MatcherAssert.assertThat;
-import static org.hamcrest.Matchers.*;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.is;
 
 
 @MicronautTest
@@ -33,6 +34,7 @@ public class ChatCompletionTest {
         ChatCompletion task = ChatCompletion.builder()
             .apiKey(this.apiKey)
             .model("gpt-3.5-turbo-0613")
+            .clientTimeout(30)
             .messages(messages)
             .build();
 

